### PR TITLE
Externalize js: case hierarchy template tag

### PR DESCRIPTION
**Pull request by orangejenny**
@calellowitz / @proteusvacuum

### CODE REVIEW

**Comment by orangejenny** [comment]
or @jmtroth0

File: corehq/apps/reports/view_helpers.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+from casexml.apps.case.views import get_wrapped_case
+from corehq.apps.hqwebapp.templatetags.proptable_tags import get_display_data
+
+
+def case_hierarchy_context(case, options):
+
+    wrapped_case = get_wrapped_case(case)
+    get_case_url = options.get('get_case_url')
+    timezone = options.get('timezone', pytz.utc)
```
Comment (proteusvacuum) [inline corehq/apps/reports/view_helpers.py]: you could maybe make these kwargs now to make it more explicit what this function is accepting

File: corehq/apps/reports/view_helpers.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        }
+
+    return get_children(case)
+
+
+def get_flat_descendant_case_list(case, get_case_url, type_info=None):
+    type_info = type_info or {}
+    hierarchy = get_case_hierarchy(case, type_info)
+    process_case_hierarchy(hierarchy, get_case_url, type_info)
+    return hierarchy['case_list']
```
Comment (proteusvacuum) [inline corehq/apps/reports/view_helpers.py]: I didn't really review this file as it looks like it is just straight copy/paste. Let me know if I should

File: corehq/ex-submodules/casexml/apps/case/templatetags/case_tags.py [hunk 3]
```diff
@@ -1,252 +1,15 @@
...
-from corehq.apps.users.util import cached_owner_id_to_display
-
-from corehq.form_processor.interfaces.dbaccessors import LedgerAccessors
-
-from casexml.apps.case.views import get_wrapped_case
 
 register = template.Library()
 
 
-def normalize_date(val):
```
Comment (proteusvacuum) [inline corehq/ex-submodules/casexml/apps/case/templatetags/case_tags.py]: :skull: :skull: :skull:

**Review by proteusvacuum — APPROVED** [review]
wahoo!

File: corehq/apps/reports/view_helpers.py (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/reports/view_helpers.py]: Good point, and doing this made me realize not all of these options are used. 839aa26

File: corehq/apps/reports/view_helpers.py (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/reports/view_helpers.py]: nope, it's indeed copy/paste

**Pull request merged**